hartshorn-core/src/main/java/org/dockbox/hartshorn/core/domain/Exceptional.java

```diff
@@ -351,7 +369,8 @@ public boolean caught() {
      * @return The {@link Exceptional}, for chaining
      * @throws NullPointerException If a value is present and {@code defaultValue} is null
      */
-    public Exceptional<T> orElse(final CheckedSupplier<T> defaultValue) {
+    @NonNull
+    public Exceptional<T> orElse(@NonNull final CheckedSupplier<T> defaultValue) {
         if (this.absent()) {
```

Comment:
Hey Diggy, I haven't added `@NonNull` to typed suppliers like these because I felt it might be going a little overboard. I can add them if you want though?

Reply:
Would be nice to have, as long as it's only done to this extent in `Exceptional`

Reply:
Do note that most return values are nullable, in this example `T` is nullable